### ISSUE TITLE
MusicGen is not running in collab

### ISSUE THREAD

**Issue by thesagarraut** [facebookresearch/audiocraft#566]
MusicGen is not running in updated collab, showing audiocraft module not found error, please guide.

**Comment by Ziegabeez** [comment]
I'd like to highlight this issue as well. I've spent hours trying different approaches to fix the dependency issue - nothing works. 

As far as I could figure out, there are at least two main conflicts arising now:

1. The audiocraft requirements file specify an old version of PyTorch (I think its 2.1.0) which is no longer available from the official PyTorch wheel index. 

2. Audiocraft also requires xformers - however latest version of xformers on PyPI has a strict requirement for PyTorch 2.8.0, so pip will remove any older versions of PyTorch and install the new one, immediately breaking audiocrafts PyTorch requirements.

There may be more issues - I'm far from an expert at dealing with these. I've tried specifying versions of everything, modifying the requirements file, trying to build my own old PyTorch wheel, etc. 


MusicGen has been such a fun and amazing tool, and I've loved running it on Colab! Thankfully I have a local environment that allows me to run it on my PC, but its incredibly slow compared to Colab. I think many of us would greatly appreciate any effort to fix this dependency issue.

Thank you for your attention to this matter!

**Comment by dorafan** [comment]
Hi, I'm experiencing several issues with installing Audiocraft.
I noticed you mentioned having a local environment; could you please share how to install it locally?

**Comment by Ziegabeez** [comment]
I didn't track exactly what I installed in the environment, and what was installed by pip as a dependency of a dependency.

I think I probably installed these:

audiocraft==1.4.0a2
torch==2.1.0+cu121
transformers==4.35.2
xformers==0.0.22.post7

Not sure how useful this is, but here is the full list from my environment. It includes many things that aren't directly related to MusicGen, and many things which were dependencies installed by pip. You should be able to paste it into your LLM and get help replicating it:

absl-py==2.3.1
aiofiles==23.2.1
aiohappyeyeballs==2.6.1
aiohttp==3.12.15
aiosignal==1.4.0
altair==5.5.0
annotated-types==0.7.0
antlr4-python3-runtime==4.9.3
anyio==4.9.0
async-timeout==5.0.1
attrs==25.3.0
audiocraft==1.4.0a2
audioread==3.0.1
av==11.0.0
blis==0.7.11
braceexpand==0.1.7
catalogue==2.0.10
certifi==2022.12.7
cffi==1.17.1
charset-normalizer==2.1.1
click==8.1.8
cloudpathlib==0.21.1
cloudpickle==3.1.1
colorama==0.4.6
colorlog==6.9.0
confection==0.1.5
contourpy==1.3.0
cycler==0.12.1
cymem==2.0.11
datasets==4.0.0
decorator==5.2.1
demucs==4.0.1
dill==0.3.8
docopt==0.6.2
dora_search==0.1.12
einops==0.8.1
encodec==0.1.1
eval_type_backport==0.2.2
exceptiongroup==1.3.0
fastapi==0.116.1
ffmpy==0.6.0
filelock==3.13.1
flashy==0.0.2
fonttools==4.59.0
frozenlist==1.7.0
fsspec==2024.6.1
ftfy==6.3.1
gitdb==4.0.12
GitPython==3.1.45
gradio==3.50.2
gradio_client==0.6.1
grpcio==1.74.0
h11==0.16.0
h5py==3.14.0
httpcore==1.0.9
httpx==0.28.1
huggingface-hub==0.33.4
hydra-colorlog==1.2.0
hydra-core==1.3.2
idna==3.4
importlib_metadata==8.7.0
importlib_resources==6.5.2
Jinja2==3.1.4
joblib==1.5.1
jsonschema==4.25.0
jsonschema-specifications==2025.4.1
julius==0.2.7
kiwisolver==1.4.7
laion_clap==1.1.7
lameenc==1.8.1
langcodes==3.5.0
language_data==1.3.0
lazy_loader==0.4
librosa==0.11.0
lightning-utilities==0.14.3
llvmlite==0.43.0
marisa-trie==1.2.1
Markdown==3.8.2
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.9.4
mdurl==0.1.2
mpmath==1.3.0
msgpack==1.1.1
multidict==6.6.4
multiprocess==0.70.16
murmurhash==1.0.13
narwhals==1.47.1
networkx==3.2.1
num2words==0.5.14
numba==0.60.0
numpy==1.26.3
omegaconf==2.3.0
openunmix==1.3.0
orjson==3.11.0
packaging==25.0
pandas==2.3.1
pesq==0.0.4
pillow==10.4.0
pip==22.0.4
platformdirs==4.3.8
pooch==1.8.2
preshed==3.0.10
primePy==1.3
progressbar==2.5
propcache==0.3.2
protobuf==6.31.1
pyarrow==21.0.0
pycparser==2.22
pydantic==2.11.7
pydantic_core==2.33.2
pydub==0.25.1
Pygments==2.19.2
pynput==1.8.1
pyparsing==3.2.3
pystoi==0.4.1
python-dateutil==2.9.0.post0
python-multipart==0.0.20
pytz==2025.2
PyYAML==6.0.2
referencing==0.36.2
regex==2024.11.6
requests==2.32.4
retrying==1.4.1
rich==14.0.0
rpds-py==0.26.0
ruff==0.12.4
safetensors==0.5.3
scikit-learn==1.6.1
scipy==1.13.1
semantic-version==2.10.0
sentencepiece==0.2.0
sentry-sdk==2.35.1
setuptools==58.1.0
shellingham==1.5.4
six==1.17.0
smart_open==7.3.0.post1
smmap==5.0.2
sniffio==1.3.1
sounddevice==0.5.2
soundfile==0.13.1
soxr==0.5.0.post1
spacy==3.7.6
spacy-legacy==3.0.12
spacy-loggers==1.0.5
srsly==2.5.1
starlette==0.47.1
submitit==1.5.3
sympy==1.13.3
tensorboard==2.20.0
tensorboard-data-server==0.7.2
thinc==8.2.5
threadpoolctl==3.6.0
tokenizers==0.15.2
tomlkit==0.12.0
torch==2.1.0+cu121
torch-audiomentations==0.12.0
torch_pitch_shift==1.2.5
torchaudio==2.1.0+cu121
torchdata==0.7.0
torchdiffeq==0.2.5
torchlibrosa==0.1.0
torchmetrics==1.7.4
torchtext==0.16.0
torchvision==0.16.0+cu121
tqdm==4.67.1
transformers==4.35.2
treetable==0.2.5
typer==0.16.0
typing_extensions==4.12.2
typing-inspection==0.4.1
tzdata==2025.2
urllib3==2.5.0
uvicorn==0.35.0
wandb==0.21.1
wasabi==1.1.3
wcwidth==0.2.13
weasel==0.4.1
webdataset==0.2.100
websockets==11.0.3
Werkzeug==3.1.3
wget==3.2
wrapt==1.17.2
xformers==0.0.22.post7
xxhash==3.5.0
yarl==1.20.1
zipp==3.23.0

**Comment by vatsal1306** [comment]
@Ziegabeez what python version are you using for running this locally?

**Comment by Ziegabeez** [comment]
Python 3.9.13
